### PR TITLE
Refactor class CrawlingInformation

### LINKED ISSUE #492: Refactor CrawlingInformation
Refactor `CrawlingInformation` to enhance the backend code quality.

### PR DESCRIPTION
**Pull request by JakobBenz**
- [x] Ensure that the commit message is [a good commit message](https://github.com/joelparkerhenderson/git_commit_message)
- [x] Ensure to use auto format in **all** files

---

## Short Description

Adjust class CrawlingInformation to GitHubCrawlingInfoDTO to conform to other DTO classes.

## Resolving Issue/ Feature

Resolves #492.

### CODE REVIEW

File: mico-core/src/main/java/io/github/ust/mico/core/dto/GitHubCrawlingInfoDTO.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    /**
+     * Link to the GitHub repo to crawl from (required).
+     */
+    private String uri;
+
+    /**
+     * The GitHub release tag. Defaults to 'latest'.
+     */
+    private String version = "latest";
```
Comment (davidkopp) [inline mico-core/src/main/java/io/github/ust/mico/core/dto/GitHubCrawlingInfoDTO.java]: Consider to use annotation `@NotEmpty` to ensure that it won't be set to null or to an empty string.

Comment (JakobBenz) [reply mico-core/src/main/java/io/github/ust/mico/core/dto/GitHubCrawlingInfoDTO.java]: Already added, commit coming in a few seconds

**Comment by JakobBenz** [comment]
My bad, CrawlingInformation in web package now deleted.

File: mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java [hunk 2]
```diff
@@ -317,9 +318,9 @@
     }
 
     @PostMapping(PATH_GITHUB_ENDPOINT)
-    public ResponseEntity<Resource<MicoService>> importMicoServiceFromGitHub(@RequestBody CrawlingInformation crawlingInformation) {
-        String uri = crawlingInformation.getUri();
-        String version = crawlingInformation.getVersion();
+    public ResponseEntity<Resource<MicoService>> importMicoServiceFromGitHub(@RequestBody GitHubCrawlingInfoDTO crawlingInfo) {
```
Comment (davidkopp) [inline mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java]: Add `@Valid` annotation

Comment (JakobBenz) [reply mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java]: Done.

**Comment by JakobBenz** [comment]
Agree. However, I think we should remove the `@NotEmpty` annotation from version since it originally was considered optional.

**Comment by davidkopp** [comment]
> Agree. However, I think we should remove the `@NotEmpty` annotation from version since it originally was considered optional. What do you think?

I don't think so. The user MUST NOT set it to `null`. If he does so, there could be a NPE. Therefore at least a `NotNull` is necessary. In addition to that `NotEmpty` makes sense because an empty string is no valid Git Tag ;)

**Comment by JakobBenz** [comment]
Ok, url probably suits the most, i will change it, however it's the name of the field since its type is `String`.

File: mico-core/src/main/java/io/github/ust/mico/core/dto/GitHubCrawlingInfoDTO.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * The GitHub release tag. Defaults to 'latest'.
+     */
+    @ApiModelProperty(extensions = {
+            @Extension(name = CustomOpenApiExtentionsPlugin.X_MICO_CUSTOM_EXTENSION, properties = {
+                    @ExtensionProperty(name = "title", value = "Version"),
+                    @ExtensionProperty(name = "x-order", value = "20"),
+                    @ExtensionProperty(name = "description", value = "The GitHub release tag. Defaults to 'latest'.") }) })
+    @NotEmpty
```
Comment (davidkopp) [inline mico-core/src/main/java/io/github/ust/mico/core/dto/GitHubCrawlingInfoDTO.java]: If you think the `@NotEmpty` could be misunderstood that it is some kind of required, add a commend like:
`Must not be {@code null} but can be omitted (default value will be used)`

File: mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java [hunk 4]
```diff
@@ -317,9 +318,9 @@
     }
 
     @PostMapping(PATH_GITHUB_ENDPOINT)
-    public ResponseEntity<Resource<MicoService>> importMicoServiceFromGitHub(@RequestBody CrawlingInformation crawlingInformation) {
-        String uri = crawlingInformation.getUri();
-        String version = crawlingInformation.getVersion();
+    public ResponseEntity<Resource<MicoService>> importMicoServiceFromGitHub(@Valid @RequestBody GitHubCrawlingInfoDTO crawlingInfo) {
+        String uri = crawlingInfo.getUrl();
```
Comment (davidkopp) [inline mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java]: You changed the DTO property `uri` to `url` but not this temporary variable `uri`

**Comment by JakobBenz** [comment]
@gepperho FYI: the DTO for crawling information has slightly changed, the field `uri` is now named `url`, which affects `POST /services/import/github/`. Also the request parameter in `GET /services/import/github/` has been renamed from `uri`to `url`.

**Comment by JanVetter** [comment]
> Side note: Sometimes the term `uri` is used and sometime `url`. I think in the context of the GitHubCrawler it would make more sense to use the term `url`.

@davidkopp @JakobBenz We always use `uri` in the `GitHubCrawler` class. Why do you want to use `url` here?

**Comment by JanVetter** [comment]
@JakobBenz **Refactoring**, as mentioned in the header of this PR, is also possible with the commands **Rename** and **Move** provided by most UIs. This would improve the traceability a lot, compared to deleting and adding new files.

**Pull request merged**